src/coreComponents/meshUtilities/InternalMeshGenerator.hpp

```diff
@@ -82,98 +39,108 @@ class InternalMeshGenerator : public MeshGeneratorBase
 
   /**
    * @brief Create a new geometric object (box, plane, etc) as a child of this group.
    * @param childKey the catalog key of the new geometric object to create
    * @param childName the name of the new geometric object in the repository
    * @return the group child
    */
-  virtual Group * createChild( string const & childKey, string const & childName ) override;
-
-  virtual void generateMesh( DomainPartition & domain ) override;
-
-  // virtual void GenerateNodesets( xmlWrapper::xmlNode const & targetNode,
-  //                                NodeManager * nodeManager ) override;
+  Group * createChild( string const & childKey, string const & childName ) override;
 
-  virtual void getElemToNodesRelationInBox ( const string & elementType,
-                                             const int index[],
-                                             const int & iEle,
-                                             int nodeIDInBox[],
-                                             const int size ) override;
+  void generateMesh( DomainPartition & domain ) override;
 
-  virtual void remapMesh ( dataRepository::Group & domain ) override;
+  void getElemToNodesRelationInBox ( const string & elementType,
+                                     const int index[],
+                                     const int & iEle,
+                                     int nodeIDInBox[],
+                                     const int size ) override;
 
-//  int m_delayMeshDeformation;
+  void remapMesh ( dataRepository::Group & domain ) override;
 
 protected:
 
-  void postProcessInput() override final;
-
-private:
+  ///@cond DO_NOT_DOCUMENT
+  struct viewKeyStruct
+  {
+    constexpr static char const * xCoordsString() { return "xCoords"; }
+    constexpr static char const * yCoordsString() { return "yCoords"; }
+    constexpr static char const * zCoordsString() { return "zCoords"; }
+    constexpr static char const * xElemsString() { return "nx"; }
+    constexpr static char const * yElemsString() { return "ny"; }
+    constexpr static char const * zElemsString() { return "nz"; }
+    constexpr static char const * xBiasString() { return "xBias"; }
+    constexpr static char const * yBiasString() { return "yBias"; }
+    constexpr static char const * zBiasString() { return "zBias"; }
+    constexpr static char const * cellBlockNamesString() { return "cellBlockNames"; }
+    constexpr static char const * elementTypesString() { return "elementTypes"; }
+    constexpr static char const * trianglePatternString() { return "trianglePattern"; }
+    constexpr static char const * meshTypeString() { return "meshType"; }
+  };
+  /// @endcond
+
+  void postProcessInput() override;
 
   /// Mesh number of dimension
   int m_dim;
+
+  /// Minimum extent of mesh dimensions
+  real64 m_min[3];
+
+  /// Maximum extent of mesh dimensions
+  real64 m_max[3];
+
+  /// Position tolerance for adding nodes to nodesets
+  real64 m_positionTolerance = 1e-10;
```

Comment:
Just put an entry like
```c++
  registerWrapper( viewKeyStruct::trianglePatternString(), &m_trianglePattern ).
    setApplyDefaultValue( 0 ).
    setInputFlag( InputFlags::OPTIONAL ).
    setDescription( "Pattern by which to decompose the hex mesh into prisms (more explanation required)" );
```
with default value and we are done.

Reply:
done